pkg/authentication/scramcredentials/scram_credentials_test.go

```diff
@@ -0,0 +1,29 @@
+package scramcredentials
+
+import (
+	"crypto/sha1"
+	"crypto/sha256"
+	"hash"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestScramSha1SecretsMatch(t *testing.T) {
+	assertSecretsMatch(t, sha1.New, "caeec61ba3b15b15b188d29e876514e8", 10, "S3cuk2Rnu/MlbewzxrmmVA==", "sYBa3XlSPKNrgjzhOuEuRlJY4dQ=", "zuAxRSQb3gZkbaB1IGlusK4jy1M=")
+	assertSecretsMatch(t, sha1.New, "4d9625b297999b3ca786d4a9622d04f1", 10, "kW9KbCQiCOll5Ljd44cjkQ==", "VJ8fFVHkPltibvT//mG/OWw44Hc=", "ceDRsgj9HezpZ4/vkZX8GZNNN50=")
+	assertSecretsMatch(t, sha1.New, "fd0a78e418dcef39f8c768222810b894", 10, "hhX6xsoID6FeWjXncuNgAg==", "TxgaZJ4cIn+S9EfTcc9IOEG7RGc=", "d6/qjwBs0qkPKfUAjSh5eemsySE=")
+}
+func TestScramSha256SecretsMatch(t *testing.T) {
+	assertSecretsMatch(t, sha256.New, "Gy4ZNMr-SYEsEpAEZv", 15000, "ajdf1E1QTsNAQdBEodB4vzQOFuvcw9K6PmouVg==", "/pBk9XBwSm9UyeQmyJ3LfogfHu9Z/XTjGmRhQDHx/4I=", "Avm8mjtMyg659LAyeD4VmuzQb5lxL5iy3dCuzfscfMc=")
+	assertSecretsMatch(t, sha256.New, "Y9SPYSJYUJB_", 15000, "Oplsu3uju+lYyX4apKb0K6xfHpmFtH99Oyk4Ow==", "oTJhml8KKZUSt9k4tg+tS6D/ygR+a2Xfo8JKjTpQoAI=", "SUfA2+SKL35u665WY5NnJJmA9L5dHu/TnWXX/0nm42Y=")
+	assertSecretsMatch(t, sha256.New, "157VDZr0h-Pz-wj72", 15000, "P/4xs3anygxu3/l2p35CSBe4Z47IV/FtE/e44A==", "jOb27nFF72SQoY7WUqKXOTR4e8jETXxMS67SONrcbjA=", "3FnslkgUweautAfPRCOEjhS+YbUYUNmdDQUGxB+oaFE=")
+	assertSecretsMatch(t, sha256.New, "P8z1sDfELCePTNbVqX", 15000, "RPNhenwTHlqW5OE597XpuwvPLaiecPpYFa58Pg==", "sJ8UhQRszLNo15cOe62+HLjt2NxmSkJGjdJpclTIMBs=", "CSg02ODAvh9+swUHoimXcDsT9lLp/A5IhQXavXl7+qA=")
+}
```

Comment:
test cases are using the salt, stored key and sever keys of users created through the mongo shell. E.g.

```bash
 db.createUser({user: "user4", 
pwd: "P8z1sDfELCePTNbVqX", 
roles: [ { role: "readWrite", db: "admin" } ]})

db.system.users.findOne({user: "user4"})

```

outputs

```bash
example-mongodb:PRIMARY> db.system.users.findOne({user: "user4"})
{
	"_id" : "admin.user4",
	"userId" : UUID("6b5aec1f-312b-491a-aeaa-b00cc374cf52"),
	"user" : "user4",
	"db" : "admin",
	"credentials" : {
		"SCRAM-SHA-1" : {
			"iterationCount" : 10000,
			"salt" : "Cd9RJHbdRt5b+kPf8YR0WQ==",
			"storedKey" : "v3BB0CLs5AwLiAgIfE/yqDujoI4=",
			"serverKey" : "yLqlYgg5GW04nAeBmFpgLU/vtto="
		},
		"SCRAM-SHA-256" : {
			"iterationCount" : 15000,
			"salt" : "RPNhenwTHlqW5OE597XpuwvPLaiecPpYFa58Pg==",
			"storedKey" : "sJ8UhQRszLNo15cOe62+HLjt2NxmSkJGjdJpclTIMBs=",
			"serverKey" : "CSg02ODAvh9+swUHoimXcDsT9lLp/A5IhQXavXl7+qA="
		}
	},
	"roles" : [
		{
			"role" : "readWrite",
			"db" : "admin"
		}
	]
}
```